src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java

```diff
@@ -0,0 +1,221 @@
+
+/**
+ * Edits the details of an existing income/expense in the address book.
+ */
+public class WalletEditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the transaction identified "
+            + "by the index number used in the displayed transaction list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_AMOUNT + "AMOUNT "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + WALLET_COMMAND_TYPE + " "
+            + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Duck Rice "
+            + PREFIX_AMOUNT + "3.80";
+
+    public static final String MESSAGE_EDIT_TRANSACTION_SUCCESS = "Edited Transaction: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditTransactionDescriptor editTransactionDescriptor;
+
+    /**
+     * @param index                     of the person in the filtered transaction list to edit
+     * @param editTransactionDescriptor details to edit the transaction with
+     */
+    public WalletEditCommand(Index index, EditTransactionDescriptor editTransactionDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editTransactionDescriptor);
+
+        this.index = index;
+        this.editTransactionDescriptor = new EditTransactionDescriptor(editTransactionDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Transaction> lastShownList = model.getFilteredTransactionList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToEdit = lastShownList.get(index.getZeroBased());
+        Transaction editedTransaction = createEditedTransaction(transactionToEdit, editTransactionDescriptor);
+
+        if (transactionToEdit instanceof Income) {
+
+            model.setIncome((Income) transactionToEdit, (Income) editedTransaction);
+        } else {
+
+            model.setExpense((Expense) transactionToEdit, (Expense) editedTransaction);
+        }
+
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_TRANSACTIONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_TRANSACTION_SUCCESS, editedTransaction));
+    }
+
+    /**
+     * Creates and returns a {@code Transaction} with the details of {@code transactionToEdit}
+     * edited with {@code editTransactionDescriptor}.
+     */
+    private static Transaction createEditedTransaction(Transaction transactionToEdit,
```

Comment:
I think you can afford to move this to the Transaction object, since modifying a Transaction might be a reusable method in the future

Reply:
Hmmm, because the `PeopleEditCommnd` have their `createdEditedPerson` inside `PeopleEditCommnd` and not `People`.

Will it be better to keep both inside the respective EditCommand or should we move both out?

Reply:
In that case let's just keep it consistent. Thanks!